### PR TITLE
Add project root and repository to project:open telemetry

### LINKED ISSUE #629: Telemetry: appland.appmap/project:open emits the absolute path of the project and the git remote url

### PR DESCRIPTION
**Pull request by dividedmind**
Fixes #629

### CODE REVIEW

File: src/lib/git.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const remotes = repo.state.remotes;
+  if (remotes.length === 0) return;
+
+  const origin = remotes.find(({ name }) => name === 'origin');
+  if (origin?.fetchUrl) return origin.fetchUrl;
+
+  const first = remotes.find(({ fetchUrl }) => !!fetchUrl);
+  if (!first?.fetchUrl) return;
+
+  return first.fetchUrl;
```
Comment (dustinbyrne) [inline src/lib/git.ts:23]: It's possible that `fetchUrl` contains basic auth credentials or access tokens in the URL. We should consider the following cases and strip out any potentially sensitive information:
```
https://$ACCESS_TOKEN@github.com/$ORG/$REPO
https://$USER:$PASSWORD@github.com/$ORG/$REPO
```

**Review by dustinbyrne — CHANGES_REQUESTED** [review]
One small request regarding credentials in the git remote URL, but otherwise it's looking good!

File: src/lib/git.ts (hunk 1 above)
Comment (dividedmind) [reply src/lib/git.ts:23]: Great point!

**Review by dustinbyrne — APPROVED** [review]

**Comment by dustinbyrne** [comment]
I rebased this before merging but [this test](https://github.com/getappmap/vscode-appland/blob/master/test/integration/authentication/appmapServerLogin.test.ts#L30) appears to be failing after the rebase with `master`.

**Comment by dividedmind** [comment]
> I rebased this before merging but [this test](https://github.com/getappmap/vscode-appland/blob/master/test/integration/authentication/appmapServerLogin.test.ts#L30) appears to be failing after the rebase with `master`.

The failure appears to be unrelated, but I fixed it anyway.

**Pull request merged**